### PR TITLE
Code Reviews

### PR DESCRIPTION
**Pull request by AurevoirXavier**
- Simplify importing behaviour
- Code cleaning
- Optimize DVM `sync_from`

### CODE REVIEW

File: node/service/src/service/mod.rs [hunk 1]
```diff
@@ -320,18 +320,14 @@ where
...
-						parent,
-					)?;
 					let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
 					let slot =
 						sp_consensus_babe::inherents::InherentDataProvider::from_timestamp_and_duration(
 							*timestamp,
 							slot_duration,
 						);
 
-					Ok((timestamp, slot, uncles))
```
Comment (HackFisher) [inline node/service/src/service/mod.rs:334]: Are we ok to remove uncles here? And why/

Request some rationale here.

**Review by HackFisher — CHANGES_REQUESTED** [review]
Rationale for removing uncles, and review its affects to validator clients backward compatibility and consensus.

See no reason to change it, and for now.

**Comment by AurevoirXavier** [comment]
After discussing with @andresilva, I think this could be removed.

See also

- [ ] https://github.com/paritytech/polkadot/pull/5750

We can wait for the Polkadot PR.

File: node/service/src/service/mod.rs (hunk 1 above)
Comment (AurevoirXavier) [reply node/service/src/service/mod.rs]: This only works on author block, like `timestamp::set`.
We can remove it here safely.

**Comment by AsceticBear** [comment]
## Pangoro

```sh
2022-07-06 18:11:52 Darwinia Runtime Module Library    
2022-07-06 18:11:52 ✌️  version 2.8.14-c81463b5-x86_64-linux-gnu    
2022-07-06 18:11:52 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2022    
2022-07-06 18:11:52 📋 Chain specification: Pangoro Development Testnet    
2022-07-06 18:11:52 🏷 Node name: Alice    
2022-07-06 18:11:52 👤 Role: AUTHORITY    
2022-07-06 18:11:52 💾 Database: RocksDb at /tmp/substrate2eNarT/chains/pangoro_dev/db/full    
2022-07-06 18:11:52 ⛓  Native runtime: Pangoro-28140 (Pangoro-0.tx0.au0)    
2022-07-06 18:11:52 `pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0    
2022-07-06 18:11:52 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:11:52 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:11:52 [0] 💸 new validator set of size 1 has been processed for era 1    
2022-07-06 18:11:52 🔨 Initializing Genesis block/state (state: 0x380c…ece5, header-hash: 0x54e0…0e6e)    
2022-07-06 18:11:52 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
2022-07-06 18:11:53 ⏱  Loaded block-time = 6s from block 0x54e0fc5d117195f6578f13dcc5d3473fae130fc54502322f24427eb815790e6e    
2022-07-06 18:11:53 👶 Creating empty BABE epoch changes on what appears to be first startup.    
2022-07-06 18:11:53 🏷 Local node identity is: 12D3KooWJMHFGu3NqCbLSpqp5RDfTWAH9bMmkZb2MjbqJxWmKBUT 
   
2022-07-06 18:11:53 DVM mapping worker starts syncing from 729781   
```

## Pangolin 

```sh
2022-07-06 18:12:43 Darwinia Runtime Module Library    
2022-07-06 18:12:43 ✌️  version 2.8.14-c81463b5-x86_64-linux-gnu    
2022-07-06 18:12:43 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2022    
2022-07-06 18:12:43 📋 Chain specification: Pangolin Development Testnet    
2022-07-06 18:12:43 🏷 Node name: Alice    
2022-07-06 18:12:43 👤 Role: AUTHORITY    
2022-07-06 18:12:43 💾 Database: RocksDb at /tmp/substratec4I6rB/chains/pangolin_dev/db/full    
2022-07-06 18:12:43 ⛓  Native runtime: Pangolin-28140 (Pangolin-0.tx0.au0)    
2022-07-06 18:12:43 `pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0    
2022-07-06 18:12:43 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:12:43 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:12:43 [0] 💸 new validator set of size 1 has been processed for era 1    
2022-07-06 18:12:43 🔨 Initializing Genesis block/state (state: 0x322b…8169, header-hash: 0x325e…ff96)    
2022-07-06 18:12:43 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
2022-07-06 18:12:44 ⏱  Loaded block-time = 6s from block 0x325ee735c02eafa376efb07b33c736edd9dece30512147a3211aa85b26e5ff96    
2022-07-06 18:12:44 👶 Creating empty BABE epoch changes on what appears to be first startup.    
2022-07-06 18:12:44 🏷 Local node identity is: 12D3KooWDQLueK5fnHHgNnfzERg3FxamoUdZ72LaA5gBGzeLT4nk    

2022-07-06 18:12:44 DVM mapping worker starts syncing from 0    

2022-07-06 18:12:44 📦 Highest known block at #0    
```

**Comment by AurevoirXavier** [comment]
Wait. I found this can not handle dev net.

**Comment by AurevoirXavier** [comment]
```
2022-07-06 18:30:32 Darwinia Runtime Module Library    
2022-07-06 18:30:32 ✌️  version 2.8.14-779c75d5-x86_64-linux-gnu    
2022-07-06 18:30:32 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2022    
2022-07-06 18:30:32 📋 Chain specification: Pangoro    
2022-07-06 18:30:32 🏷 Node name: Alice    
2022-07-06 18:30:32 👤 Role: AUTHORITY    
2022-07-06 18:30:32 💾 Database: RocksDb at /tmp/substrate7UIzgI/chains/pangoro/db/full    
2022-07-06 18:30:32 ⛓  Native runtime: Pangoro-28140 (Pangoro-0.tx0.au0)    
2022-07-06 18:30:32 🔨 Initializing Genesis block/state (state: 0xbd47…6ac3, header-hash: 0xf37f…db58)    
2022-07-06 18:30:32 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
2022-07-06 18:30:32 ⏱  Loaded block-time = 6s from block 0xf37f31c4481e3c937ad40b672c561d0c0b1fb70b40e51a0d56dad30d60e4db58    
2022-07-06 18:30:32 👶 Creating empty BABE epoch changes on what appears to be first startup.    
2022-07-06 18:30:32 🏷 Local node identity is: 12D3KooWGQsrJC8GsUTYeYkEA9YtH1gG93dnW2mCkev1oXjUBxZt    
2022-07-06 18:30:32 DVM mapping worker starts syncing from 729781


2022-07-06 18:30:58 Darwinia Runtime Module Library    
2022-07-06 18:30:58 ✌️  version 2.8.14-779c75d5-x86_64-linux-gnu    
2022-07-06 18:30:58 ❤️  by Darwinia Network <hello@darwinia.network>, 2018-2022    
2022-07-06 18:30:58 📋 Chain specification: Pangoro Development Testnet    
2022-07-06 18:30:58 🏷 Node name: Alice    
2022-07-06 18:30:58 👤 Role: AUTHORITY    
2022-07-06 18:30:58 💾 Database: RocksDb at /tmp/substratemMrTWk/chains/pangoro_dev/db/full    
2022-07-06 18:30:58 ⛓  Native runtime: Pangoro-28140 (Pangoro-0.tx0.au0)    
2022-07-06 18:30:58 `pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0    
2022-07-06 18:30:58 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:30:58 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-07-06 18:30:58 [0] 💸 new validator set of size 1 has been processed for era 1    
2022-07-06 18:30:58 🔨 Initializing Genesis block/state (state: 0xb61d…22fb, header-hash: 0x0d14…874e)    
2022-07-06 18:30:58 👴 Loading GRANDPA authority set from genesis on what appears to be first startup.    
2022-07-06 18:30:59 ⏱  Loaded block-time = 6s from block 0x0d14d92bb2ab2eb594551c62d2de6adca8baadf79e4df2f6b3579632e29f874e    
2022-07-06 18:30:59 👶 Creating empty BABE epoch changes on what appears to be first startup.    
2022-07-06 18:30:59 🏷 Local node identity is: 12D3KooWQTssn5bpn73LpLoEfvhLFLkkbdJrGhEWVMEtu9rXMLxG    
2022-07-06 18:30:59 DVM mapping worker starts syncing from 0
```

**Pull request merged**